llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py

```diff
@@ -142,33 +142,39 @@ def _get_site_id_with_host_name(self, access_token, sharepoint_site_name) -> str
 
         self._authorization_headers = {"Authorization": f"Bearer {access_token}"}
 
-        response = requests.get(
-            url=site_information_endpoint,
-            headers=self._authorization_headers,
-        )
-
-        if response.status_code == 200 and "value" in response.json():
-            if (
-                len(response.json()["value"]) > 0
-                and "id" in response.json()["value"][0]
-            ):
-                # find the site with the specified name
-                for site in response.json()["value"]:
-                    if site["name"].lower() == sharepoint_site_name.lower():
-                        return site["id"]
+        while site_information_endpoint:
+            response = requests.get(
+                url=site_information_endpoint,
+                headers=self._authorization_headers,
+            )
 
-                raise ValueError(
-                    f"The specified sharepoint site {sharepoint_site_name} is not found."
-                )
+            json_response = response.json()
+            if response.status_code == 200 and "value" in json_response:
+                if (
+                    len(json_response["value"]) > 0
+                    and "id" in json_response["value"][0]
+                ):
+                    # find the site with the specified name
+                    for site in json_response["value"]:
+                        if site["name"].lower() == sharepoint_site_name.lower():
+                            print(f"looks like i found the site {sharepoint_site_name}")
```

Comment:
Let's remove this debug print

Reply:
Ahh, damn..  Thought I cleaned that up.  Will do.